src/docs/product/session-replay/performance-overhead.mdx

```diff
@@ -24,9 +24,36 @@ Sentry's Session Replay SDK takes several measures to avoid negatively impacting
 - The SDK is built to gracefully downgrade when needed. In the event several thousand DOM mutations happen at once, the SDK will disable recording to make sure recording the website doesn't further degrade user experience. The [mutation limit values are configurable](/platforms/javascript/session-replay/configuration/#mutation-limits).
 
-## Performance Overhead in Sentry
+## Factors in Performance Overhead
 
-While the performance overhead of our client SDK will be imperceptible to end-users most of the time, it can vary based on the complexity of your application. If an applications has a large DOM and numerous DOM node mutations, it will have a higher overhead than simpler, mostly static sites. The only way to get accurate metrics is to measure performance overhead yourself.
+While the performance overhead of our client SDK will be imperceptible to end-users most of the time, it can vary based on the complexity of your application. If an application has a large DOM and numerous DOM node mutations, there will be higher overhead compared to simpler, mostly static sites.
+
+There are different stages in the lifecycle of a webpage, each with different performance characteristics. Here is a list of stages in the lifecycle, and the most important performance metrics to be aware of.
+
+### Pageload stage
+
+- [SDK bundle size is something we optimize for.](https://blog.sentry.io/sentry-bundle-size-how-we-reduced-replay-sdk-by-35/) Make sure you have tree-shaking enabled, and have updated to the latest SDK version (at least version 7.78.0) for the smallest bundle sizes.
+- The size of your initial HTML page affects how long the SDK takes to read the initial snapshot. Using your browser devtools, filter the Network panel to show "type=document" and look at the size to get a sense of the impact on your site. A sterotypical single-page-app that shows a loading screen and needs to wait for JavaScript to render everything will have negligible overhead from the initial snapshot.
+- All Replay SDK data is compressed before being sent across the network. HTML is highly compressable, we were surprised how effective compression is on real websites. Network overhead is further reduced if you have [PII Masking and Blocking](https://docs.sentry.io/platforms/javascript/session-replay/privacy/) enabled (which is the default). Masked text is replaced with `*` which gzip shrinks really well, while blocked content is not included at all.
+
+### Scrolling & reading stage
```

Comment:
not sure if I understand this heading, what does "reading stage" refer to here? Maybe something like this would be clearer (if I understand this correctly):

```suggestion
### Capturing mouse/touch events
```